crawler/fundcode_collector.go

```diff
@@ -0,0 +1,79 @@
+package crawler
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/gocolly/colly/v2"
+	"github.com/robertkrimen/otto"
+	log "github.com/sirupsen/logrus"
+)
+
+func toBasicFund(elements []*otto.Value) (*BasicFund, error) {
+	if len(elements) != 5 {
+		return nil, errors.New("toBasicFund Error: len(elements) != 5")
+	}
+	for i, e := range elements {
+		if !e.IsString() {
+			return nil, errors.New(fmt.Sprintf("toBasicFund Error: element(%d) is not string", i))
+		}
+	}
+	bf := &BasicFund{
+		ID:        elements[0].String(),
+		ShortCode: elements[1].String(),
+		Type:      elements[2].String(),
+		NameCN:    elements[3].String(),
+		NamePY:    elements[4].String(),
+	}
+	return bf, nil
+}
+
+func (cl *crawlerImpl) GetAllBasicFund(sync bool) {
+	log.Debugf("Start collecting all basic fund")
+	c, ok := cl.mCollectors[basicFundCollector]
+	if !ok {
+		vm := otto.New()
```

Comment:
it seems like OnResponse could be invoked concurrently, put this declaration inside the handler.